src/rx-query.ts

```diff
@@ -1,51 +1,41 @@
+import {BehaviorSubject, firstValueFrom, merge, Observable} from 'rxjs';
+import {distinctUntilChanged, filter, map, mergeMap, shareReplay, startWith} from 'rxjs/operators';
 import {
-    BehaviorSubject,
-    firstValueFrom,
-    Observable,
-    merge
-} from 'rxjs';
-import {
-    mergeMap,
-    filter,
-    map,
-    startWith,
-    distinctUntilChanged,
-    shareReplay
-} from 'rxjs/operators';
-import {
-    sortObject,
-    stringifyFilter,
-    pluginMissing,
-    overwriteGetterForCaching,
+    areRxDocumentArraysEqual,
+    ensureNotFalsy,
     now,
-    PROMISE_RESOLVE_FALSE,
+    overwriteGetterForCaching,
+    pluginMissing,
+    PROMISE_RESOLVE_FALSE, RX_META_LWT_MINIMUM,
     RXJS_SHARE_REPLAY_DEFAULTS,
-    ensureNotFalsy,
-    areRxDocumentArraysEqual
+    sortObject,
+    stringifyFilter
 } from './plugins/utils';
-import {
-    newRxError
-} from './rx-error';
-import {
-    runPluginHooks
-} from './hooks';
+import {newRxError} from './rx-error';
+import {runPluginHooks} from './hooks';
 import type {
-    RxCollection,
-    RxDocument,
-    RxQueryOP,
-    RxQuery,
     MangoQuery,
-    MangoQuerySortPart,
     MangoQuerySelector,
+    MangoQuerySortPart,
     PreparedQuery,
+    QueryMatcher,
     RxChangeEvent,
-    RxDocumentWriteData,
+    RxCollection,
+    RxDocument,
     RxDocumentData,
-    QueryMatcher
+    RxDocumentWriteData,
+    RxQuery,
+    RxQueryOP
 } from './types';
-import { calculateNewResults } from './event-reduce';
-import { triggerCacheReplacement } from './query-cache';
-import { getQueryMatcher, normalizeMangoQuery } from './rx-query-helper';
+import {calculateNewResults} from './event-reduce';
+import {triggerCacheReplacement} from './query-cache';
+import {getQueryMatcher, getSortComparator, normalizeMangoQuery} from './rx-query-helper';
+import {murmurHash} from 'ohash';
+
+export interface QueryCacheBackend {
+    getItem<T extends string | string[]>(key: string): Promise<T | null>;
+    setItem<T extends string | string[]>(key: string, value: T): Promise<T>;
```

Comment:
do you even need the T type param here? isn't it always just a cache for `string | string[]`?

Reply:
IIRC I did this to be compatible with the `localforage` interface. There are probably nicer ways to write this.

Reply:
OK